### DISCUSSION TITLE
Can't catch an exception when port is busy

### DISCUSSION THREAD

**Discussion by avoitenko** [dotnet/WatsonWebserver#67] [Q&A]
Hello. I cannot catch an exception when I call:
`server.Start(11212);`
and this port 11212 is busy at the moment.
I need to raice an exception or Start() should return False.
Any suggestions.

**Comment by jchristn** [comment]
Hi @avoitenko I moved the HttpListener.Start() into the ```Start()``` and ```StartAsync()``` methods.  That should allow you to catch any exceptions associated with a port being already in use.

NuGet: https://www.nuget.org/packages/Watson/4.2.0.2
Commit: https://github.com/jchristn/WatsonWebserver/commit/7f675e4a83268a62065617bc7562a7dabefb4e63

Please let me know if there are any issues.

**Reply by avoitenko** [reply]
@jchristn Thanks. It works.
Please do the same fix for WatsonTcp.

**Reply by jchristn** [reply]
Done, thanks for the suggestion
